src/frontend/next/src/components/SearchPage.tsx

```diff
@@ -38,11 +36,13 @@ const SearchPage = () => {
   // Form was submitted, so go ahead and sync to URL, (re)triggering search.
   function onSubmitHandler(event: FormEvent) {
     event.preventDefault();
-    setTextParam(text);
-    setFilterParam(filter);
     router.push(`/search?text=${text}&filter=${filter}`);
   }
 
+  useEffect(() => {
+    setText(textParam);
+    setFilter(filterParam);
+  }, [textParam, filterParam]);
```

Comment:
Add a black line below this, before the `return (`

Reply:
It has been updated.